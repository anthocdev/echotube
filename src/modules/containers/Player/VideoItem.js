import React from "react";
import Card from "react-bootstrap/Card";

export default function VideoItem({ index, video, selectedVideo }) {
  return (
    <div
      onClick={() => {
        selectedVideo(index, video.VideoLink);
      }}
    >
      <Card className="videoCard">
        <Card.Img
          variant="top"
          src={`https://i.ytimg.com/vi/${video.VideoLink}/hqdefault.jpg`}
        />
        <Card.ImgOverlay>
          <Card.Title>{video.Name}</Card.Title>
        </Card.ImgOverlay>
      </Card>
    </div>
  );
}
