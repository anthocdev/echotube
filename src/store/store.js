import { applyMiddleware, createStore, compose } from "redux";
import { createLogger } from "redux-logger";
import { default as thunk } from "redux-thunk";
import { createPromise } from "redux-promise-middleware";
import axios from "axios";

import reducer from "../reducers";
//Pulling JWT Token from local storage
const jwttoken = localStorage.getItem("JWT_TOKEN");
//Setting JWT Token as common header for axios
axios.defaults.headers.common["Authorization"] = jwttoken;
//Create store with middleware included
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; //Adding redux dev tool compatibility
const middleware = applyMiddleware(createPromise(), thunk, createLogger()); //Adding thunk and console logging for testing as middleware

export default createStore(
  reducer,
  {
    //Initializing auth state if JWT token is available
    auth: {
      jwtToken: jwttoken,
      isAuth: jwttoken ? true : false
    }
  },
  composeEnhancers(middleware)
);
