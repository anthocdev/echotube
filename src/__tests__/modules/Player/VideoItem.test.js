import React from "react";
import ReactDOM from "react-dom";
import VideoItem from "../../../modules/containers/Player/VideoItem";
import { Provider } from "react-redux";
import store from "../../../store/store";

/* Example Video Object for Integration Testing with API Data Structure */
const data = {
  VideoID: 27,
  Name: "Castion - El Zorro",
  Description:
    "[Electro House] Castion - El Zorro (Extended Mix) ➥ Our Spotify: https://illumi.fanlink.to/spotify 📱 Subscribe! Help us to 250,000 🔥 Turn on notifications to stay updated! 🔔 El Zorro by Castion is out now on Mixmash Bold! 🔥 ⬇️ Stream/Download: http://bit.ly/ELZORRO 🎶 Our Spotify Playlists: Future House 🕺 https://illumi.fanlink.to/futurehouse Future Bounce 🔥 https://illumi.fanlink.to/futurebounce Bass House 😈 https://illumi.fanlink.to/basshouse Deep House 🌴 https://illumi.fanlink.to/deephouse YouTube Uploads ▶️ https://illumi.fanlink.to/youtubeuploads 🙏 Join to Illumi Nation Discord: https://discord.gg/Qd8KXru 👕 Merch: http://bit.ly/illumimerch 🎧 Submit Your Track: https://submissions.illumimusic.eu 📀 Promo for Record Labels: promo@illumimusic.eu 📧 Contact: contact@illumimusic.eu ©️🚫This track is NOT copyright free! Copyright Free Music here: http://bit.ly/2si4AyM ➥ Follow Illumi Music: https://www.facebook.com/IllumiMusic https://soundcloud.com/illumi-music https://twitter.com/IllumiMusic https://www.instagram.com/illumimusicyt https://www.snapchat.com/add/illumimusic https://vk.com/illumimusictv ➥ Follow Castion: https://soundcloud.com/castionmusic https://www.facebook.com/CastionMusic https://twitter.com/CastionMusic https://www.instagram.com/castionmusic 📷 Photo by Philipp Lansing Tags: #Castion #ElZorro #BassHouse #Mixmash #FutureHouse #FutureBounce #HouseMusic #DeepHouse #ElectroHouse #EDM #PartyMusic #ShuffleDance #Shuffle #FestivalMusic #Electronic #ElectronicMusic #Dance #Dancemusic #Summermusic #Music #NewMusic #ElectronicDanceMusic #IllumiMusic #Illumi #Music #Future #House Castion - El Zorro Castion - El Zorro (Extended Mix)",
  ChannelName: "Illumi Music",
  ChannelID: "UCXarSAuECvnQFBb1vMcMsjQ",
  VideoLink: "DZjpDfgXNe0",
};

describe("Individual Video Item Module Testing", () => {
  it("Renders Successfully with Current Structure", () => {
    const div = document.createElement("div");

    ReactDOM.render(
      <Provider store={store}>
        <VideoItem video={data} key={1} index={1} selectedVideo={null} />
      </Provider>,
      div
    );
  });
});
