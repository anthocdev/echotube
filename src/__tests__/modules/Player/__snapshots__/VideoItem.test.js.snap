// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Individual Video Item Module Testing Renders Matches Snapshot 1`] = `
<div
  className="makeStyles-root-1"
  onClick={[Function]}
  style={
    Object {
      "borderColor": "white",
    }
  }
>
  <div
    className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "borderColor": "white",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root makeStyles-preview-3"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <img
            alt="complex"
            className="makeStyles-img-4"
            src="https://i.ytimg.com/vi/DZjpDfgXNe0/hqdefault.jpg"
          />
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-true"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
          >
            <h6
              className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
            >
              Castion - El Zorro
            </h6>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <p
              className="MuiTypography-root MuiTypography-body2"
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Illumi Music
            </p>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-bigIcon-6"
            focusable="false"
            style={
              Object {
                "cursor": "pointer",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
